client/allocrunner/task_hook_coordinator_test.go

```diff
@@ -93,6 +93,53 @@ func TestTaskHookCoordinator_MainRunsAfterPrestart(t *testing.T) {
 	require.Truef(t, isChannelClosed(sideCh), "%s channel was open, should be closed", sideTaskName)
 }
 
+func TestTaskHookCoordinator_MainRunsAfterManyInitTasks(t *testing.T) {
+	logger := testlog.HCLogger(t)
+
+	alloc := mock.Alloc()
+	tasks := alloc.Job.TaskGroups[0].Tasks
+	init1 := initTask()
+	init1.Name = "init-1"
+	init2 := initTask()
+	init2.Name = "init-2"
+
+	tasks = append(tasks, init1)
+	tasks = append(tasks, init2)
+
+	coord := newTaskHookCoordinator(logger, tasks)
+	mainCh := coord.startConditionForTask(tasks[0])
+	init1Ch := coord.startConditionForTask(tasks[1])
+	init2Ch := coord.startConditionForTask(tasks[2])
+
+	require.Falsef(t, isChannelClosed(mainCh), "%s channel was closed, should be open", tasks[0].Name)
+	require.Truef(t, isChannelClosed(init1Ch), "%s channel was open, should be closed", tasks[1].Name)
+	require.Truef(t, isChannelClosed(init2Ch), "%s channel was open, should be closed", tasks[2].Name)
```

Comment:
very nit picky: we can improve the message a bit - the `isChannelClosed` part already captures that part.  Maybe explain intent instead of the actual mechanics, e.g. `"%s task should be clear to run but isn't"`